src/libraries/TickBitmap.sol

```diff
@@ -28,11 +28,32 @@ library TickBitmap {
     /// @param tick The tick to flip
     /// @param tickSpacing The spacing between usable ticks
     function flipTick(mapping(int16 => uint256) storage self, int24 tick, int24 tickSpacing) internal {
-        unchecked {
-            if (tick % tickSpacing != 0) revert TickMisaligned(tick, tickSpacing); // ensure that the tick is spaced
-            (int16 wordPos, uint8 bitPos) = position(tick / tickSpacing);
-            uint256 mask = 1 << bitPos;
-            self[wordPos] ^= mask;
+        /**
+         * Equivalent to the following Solidity:
+         *     if (tick % tickSpacing != 0) revert TickMisaligned(tick, tickSpacing);
+         *     (int16 wordPos, uint8 bitPos) = position(tick / tickSpacing);
+         *     uint256 mask = 1 << bitPos;
+         *     self[wordPos] ^= mask;
+         */
+        /// @solidity memory-safe-assembly
+        assembly {
+            // ensure that the tick is spaced
+            if smod(tick, tickSpacing) {
+                mstore(0, 0xd4d8f3e6) // selector for TickMisaligned(int24,int24)
```

Comment:
i see we have no tests that revert for this in TickBitmap.t.sol. Please can you add one?
```
vm.expectRevert(abi.encodeWithSelector(TickBitmap.TickMisaligned.selector, tickLower, tickSpacing));
```

That way we always verify this selector is right too

Reply:
It's included in the fuzz test I added. Do you still want a separate one?
https://github.com/Uniswap/v4-core/pull/653/files#diff-4583698d7e46719feb044f179651f012d50645cd55e43405a959c2dea73ae751R104-R105

Reply:
yeah youre right thats fine, idk why the fuzz wasnt loading earlier haha